src/query/Index.hpp

```diff
@@ -15,6 +15,7 @@ namespace query {
 
 /// Struct to hold index and distance information of the closest primitive
 template <class Tag>
+// TODO: Initialize these values with defaults
```

Comment:
Maybe it would be good to have something like `static int constants::invalidIndex = std::numeric_limits<int>::min()` in a utilities function in order to use it for such places and consistently throughout the code. That's actually similar to what the deal people do.

Reply:
It would also enable to catch up invalid indices in the code.

Reply:
This is strongly related to #248.
My suggestion is to simply use `-1` now and come up with a solution later.
You can use the doxygen `/// @todo text` to create an entry in https://precice.org/doxygen/develop/todo.html

Reply:
Added it.